From 1622a5c9c2f123ef27fc3a52938162a68892e725 Mon Sep 17 00:00:00 2001
From: Scott Talbert <swt@techie.net>
Date: Thu, 1 Feb 2024 20:36:50 -0500
Subject: [PATCH] Fix WebView tests with WebKitGTK 2.43+

It seems that WebKitGTK is now failing to navigate to about: URLs unless
they are about:blank or about:srcdoc, so use about:srcdoc as the
alternate URL to fix the WebView tests.

Ref: https://github.com/WebKit/WebKit/commit/3c3163e71f647db507949ecebad35d0f2ffb33f3
---
 tests/controls/webtest.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/controls/webtest.cpp b/tests/controls/webtest.cpp
index c7055d895f07..3f1ad5e0ca06 100644
--- a/tests/controls/webtest.cpp
+++ b/tests/controls/webtest.cpp
@@ -45,7 +45,11 @@ class WebViewTestCase
         }
         else
 #endif
+#if wxUSE_WEBVIEW_WEBKIT2
+            m_alternateHistoryURL = "about:srcdoc";
+#else
             m_alternateHistoryURL = "about:";
+#endif
     }
 
     ~WebViewTestCase()
